torchmetrics/classification/ranking.py

```diff
@@ -0,0 +1,173 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.functional.classification.ranking import (
+    _coverage_error_compute,
+    _coverage_error_update,
+    _label_ranking_average_precision_compute,
+    _label_ranking_average_precision_update,
+    _label_ranking_loss_compute,
+    _label_ranking_loss_update,
+)
+from torchmetrics.metric import Metric
+
+
+class CoverageError(Metric):
+    """Computes multilabel coverage error. The score measure how far we need to go through the ranked scores to
+    cover all true labels. The best value is equal to the average number of labels in the target tensor per sample.
+
+    Args:
+        kwargs:
+            Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+
+    Example:
+        >>> from torchmetrics import CoverageError
+        >>> _ = torch.manual_seed(42)
+        >>> preds = torch.rand(10, 5)
+        >>> target = torch.randint(2, (10, 5))
+        >>> metric = CoverageError()
+        >>> metric(preds, target)
+        tensor(3.9000)
+    """
+
+    higher_is_better: bool = False
+    is_differentiable: bool = False
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.add_state("coverage", torch.tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("numel", torch.tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("weight", torch.tensor(0.0), dist_reduce_fx="sum")
+
+    def update(self, preds: Tensor, target: Tensor, sample_weight: Optional[Tensor] = None) -> None:  # type: ignore
+        """
+        Args:
+            preds: tensor of shape ``[N,L]`` where ``N`` is the number of samples and ``L`` is the number
+                of labels. Should either be probabilities of the positive class or corresponding logits
+            target: tensor of shape ``[N,L]`` where ``N`` is the number of samples and ``L`` is the number
+                of labels. Should only contain binary labels.
+            sample_weight: tensor of shape ``N`` where ``N`` is the number of samples. How much each sample
+                should be weighted in the final score.
+        """
+        coverage, numel, sample_weight = _coverage_error_update(preds, target, sample_weight)
+        self.coverage += coverage
+        self.numel += numel
+        if sample_weight is not None:
+            self.weight += sample_weight
+
+    def compute(self) -> Tensor:
+        """Computes the multilabel coverage error."""
+        return _coverage_error_compute(self.coverage, self.numel, self.weight)
+
+
+class LabelRankingAveragePrecisionScore(Metric):
```

Comment:
```suggestion
class LabelRankingAveragePrecision(Metric):
```
just thinking is the Score is needed?